ping/ping.c

```diff
@@ -57,6 +57,9 @@
 #include <ifaddrs.h>
 #include <math.h>
 
+/* FIXME: global_rts will be removed in future */
+struct ping_rts *global_rts;
+
```

Comment:
What is actually wrong with having it as a global variable? Once the parameters are packed in a structure it solves the problems you describe with the commit description nicely and passing this pointer to all functions complicates things for no good reason. So unless you plan to pass a different pointer to different functions at runtime it makes more sense to keep it as a global variable.

Reply:
Moving toward better scoping is best to by having some scoping, how ever little it might be at the beginning.

Reply:
I would really use static struct until it's needed.

Reply:
Passing the struct conditionally happens already. Meanwhile since it was required to make this change bigger to make a point I really need this runtime structure to make overall program structure better here is next commit 99aa4091b8cd003f259a668aa302106594085470

p.s. There were couple bugs in global runtime struct commit that should be fixed.